core/GlobalState.cc

```diff
@@ -1755,13 +1773,26 @@ unique_ptr<GlobalState> GlobalState::deepCopy(bool keepId) const {
     result->onlyErrorClasses = this->onlyErrorClasses;
     result->dslPlugins = this->dslPlugins;
     result->dslRubyExtraArgs = this->dslRubyExtraArgs;
-    result->names.reserve(this->names.capacity());
+    result->utf8Names.reserve(this->utf8Names.capacity());
+    result->constantNames.reserve(this->constantNames.capacity());
+    result->uniqueNames.reserve(this->uniqueNames.capacity());
     if (keepId) {
-        result->names.resize(this->names.size());
-        ::memcpy(result->names.data(), this->names.data(), this->names.size() * sizeof(Name));
+        result->utf8Names.resize(this->utf8Names.size());
+        ::memcpy(result->utf8Names.data(), this->utf8Names.data(), this->utf8Names.size() * sizeof(UTF8Name));
```

Comment:
Does this use `memcpy` for performance reasons? I wonder how different it would be with `std::copy_n` instead.

Reply:
As far as I can tell, we only use this in the fuzzer for performance.